pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -427,33 +440,187 @@ func (c *cyclonedxParser) getAffectedPackages(ctx context.Context, vulnInput *mo
-			logger.Debugf("[cdx vex] package range versions not supported yet: %q", affect.Range)
+			purls, err := c.findCDXPkgVersionIDs(ctx, pkgIdentifier, affect.Range)
+			if err != nil {
+				return nil, fmt.Errorf("error searching version ranges for %s: %v", affect.Range, err)
+			}
+
+			for _, purl := range purls {
+				var addVulnDataErr error
+				viList, addVulnDataErr = c.AddVulnerabilityIngestionData(purl, vexData, vulnInput, viList)
+				if addVulnDataErr != nil {
+					return nil, fmt.Errorf("could not add Vulnerability information for %s: %v", pkgIdentifier, err)
+				}
+			}
+
 			continue
 		}
 		if affect.Version == "" {
 			return nil, fmt.Errorf("no version found for package ref %q", pkgRef)
 		}
-		vi := &assembler.VexIngest{
-			VexData:       &vexData,
-			Vulnerability: vulnInput,
-		}
 
 		// create guac specific identifier string using affected package name and version.
-		pkgID := guacCDXPkgPurl(pkdIdentifier, affect.Version, "", false)
-		pkg, err := asmhelpers.PurlToPkg(pkgID)
+		purl := guacCDXPkgPurl(pkgIdentifier, affect.Version, "", false)
+		var err error
+		viList, err = c.AddVulnerabilityIngestionData(purl, vexData, vulnInput, viList)
 		if err != nil {
-			return nil, fmt.Errorf("unable to create package input spec from guac pkg purl: %v", err)
+			return nil, fmt.Errorf("could not add Vulnerability information for %s: %v", pkgIdentifier, err)
 		}
-		vi.Pkg = pkg
-		viList = append(viList, *vi)
-		c.identifierStrings.PurlStrings = append(c.identifierStrings.PurlStrings, pkgID)
 	}
 
 	return &viList, nil
 }
 
+func (c *cyclonedxParser) AddVulnerabilityIngestionData(purl string, vexData model.VexStatementInputSpec, vulnInput *model.VulnerabilityInputSpec, viList []assembler.VexIngest) ([]assembler.VexIngest, error) {
+	pkg, err := asmhelpers.PurlToPkg(purl)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create package input spec from guac pkg purl: %v", err)
+	}
+	vi := &assembler.VexIngest{
+		VexData:       &vexData,
+		Vulnerability: vulnInput,
+		Pkg:           pkg,
+	}
+	viList = append(viList, *vi)
+	c.identifierStrings.PurlStrings = append(c.identifierStrings.PurlStrings, purl)
+	return viList, nil
+}
+
+func (c *cyclonedxParser) findCDXPkgVersionIDs(ctx context.Context, pkgIdentifier string, versionRange string) ([]string, error) {
+	var matchingVersionPurls []string
+
+	// search for all packages that have a matching type, namespace, and name.
+	typeGUAC, namespace := "guac", "pkg"
+	pkgResponse, err := getPackages(ctx, c.gqlClient, model.PkgSpec{
+		Type:      &typeGUAC,
+		Namespace: &namespace,
+		Name:      &pkgIdentifier,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error querying for packages1: %w", err)
+	}
+
+	pkgVersionsMap := map[string]string{}
+	var pkgVersions []string
+	for _, depPkgVersion := range pkgResponse.Packages[0].Namespaces[0].Names[0].Versions {
+		pkgVersions = append(pkgVersions, depPkgVersion.Version)
+		pkgVersionsMap[depPkgVersion.Version] = depPkgVersion.Id
```

Comment:
not sure i understand this why are you replacing the value in the map twice in line 509 and again in 510?

Reply:
That was something I, by mistake, left from testing. Thanks!